csharp/ql/src/semmle/code/csharp/ir/implementation/raw/internal/TranslatedCondition.qll

```diff
@@ -0,0 +1,251 @@
+}
+
+class TranslatedParenthesisCondition extends TranslatedFlexibleCondition {
+  override ParenthesizedExpr expr;
+
+  final override Instruction getChildTrueSuccessor(TranslatedCondition child) {
+    child = getOperand() and
+    result = getConditionContext().getChildTrueSuccessor(this)
+  }
+
+  final override Instruction getChildFalseSuccessor(TranslatedCondition child) {
+    child = getOperand() and
+    result = getConditionContext().getChildFalseSuccessor(this)
+  }
+
+  final override TranslatedCondition getOperand() {
+    result = getTranslatedCondition(expr.getExpr())
+  }
+}
+
+class TranslatedNotCondition extends TranslatedFlexibleCondition {
+  override LogicalNotExpr expr;
+
+  override Instruction getChildTrueSuccessor(TranslatedCondition child) {
+    child = getOperand() and
+    result = getConditionContext().getChildFalseSuccessor(this)
+  }
+
+  override Instruction getChildFalseSuccessor(TranslatedCondition child) {
+    child = getOperand() and
+    result = getConditionContext().getChildTrueSuccessor(this)
+  }
+  
+  override TranslatedCondition getOperand() {
+    result = getTranslatedCondition(expr.getOperand())
+  }
+}
+
+abstract class TranslatedNativeCondition extends TranslatedCondition,
+  TTranslatedNativeCondition {
+  TranslatedNativeCondition() {
+    this = TTranslatedNativeCondition(expr)
+  }
+
+  override final Instruction getChildSuccessor(TranslatedElement child) {
+    none()
+  }
+}
+
+abstract class TranslatedBinaryLogicalOperation extends
+  TranslatedNativeCondition, ConditionContext {
+  override BinaryLogicalOperation expr;
+
+  override final TranslatedElement getChild(int id) {
+    id = 0 and result = getLeftOperand() or
+    id = 1 and result = getRightOperand()
+  }
+
+  override final Instruction getFirstInstruction() {
+    result = getLeftOperand().getFirstInstruction()
+  }
+
+  override final predicate hasInstruction(Opcode opcode, InstructionTag tag,
+    Type resultType, boolean isLValue) {
+    none()
+  }
+
+  override final Instruction getInstructionSuccessor(InstructionTag tag,
+    EdgeKind kind) {
+    none()
+  }
+
+  final TranslatedCondition getLeftOperand() {
+    result = getTranslatedCondition(expr.getLeftOperand())
+  }
+
+  final TranslatedCondition getRightOperand() {
+    result = getTranslatedCondition(expr.getRightOperand())
+  }
+}
+
+class TranslatedLogicalAndExpr extends TranslatedBinaryLogicalOperation {
+  TranslatedLogicalAndExpr() {
+    expr instanceof LogicalAndExpr
+  }
+
+  override Instruction getChildTrueSuccessor(TranslatedCondition child) {
+    (
+      child = getLeftOperand() and
+      result = getRightOperand().getFirstInstruction()
+    ) or
+    (
+      child = getRightOperand() and
+      result = getConditionContext().getChildTrueSuccessor(this)
+    )
+  }
+
+  override Instruction getChildFalseSuccessor(TranslatedCondition child) {
+    (child = getLeftOperand() or child = getRightOperand()) and
```

Comment:
Is there `getAnOperand()`? If not it might be quite good to add.

Reply:
Added.